mongo/integration/database_test.go

```diff
@@ -249,20 +249,11 @@ func TestDatabase(t *testing.T) {
 			assert.Nil(mt, err, "expected command %s to contain key 'batchSize'", evt.Command)
 		})
 		mt.RunOpts("authorizedCollections", mtest.NewOptions().MinServerVersion("4.0"), func(mt *mtest.T) {
-			// Create two new collections so there will be three total.
-			createCollections(mt, 2)
-
 			mt.ClearEvents()
 			lcOpts := options.ListCollections().SetAuthorizedCollections(true)
-			cur, err := mt.DB.ListCollections(mtest.Background, bson.D{}, lcOpts)
+			_, err := mt.DB.ListCollections(mtest.Background, bson.D{}, lcOpts)
 			assert.Nil(mt, err, "ListCollections error: %v", err)
 
-			var numColls int
-			for cur.Next(mtest.Background) {
-				numColls++
-			}
-			assert.Equal(mt, 3, numColls, "expected 3 collections to be returned, got %v", numColls)
```

Comment:
I'm not sure we can actually assert the number of collections returned here; it seems to depend on the configuration of the instance we're testing against (either 5, 6 or 7). 

It'd be great to create a collection that we don't have access to and assert that it's not included in the returned "authorized" collections, but that doesn't seem trivial and might just be testing server logic. Let me know what you think.

Reply:
I think asserting that the `authorizedCollections` option is present in the command started events is sufficient.